spec/models/amusement_park_spec.rb

```diff
@@ -3,5 +3,30 @@
 RSpec.describe AmusementPark, type: :model do
   describe "relationships" do
     it { should have_many(:rides) }
+    it { should have_many(:mechanics).through(:rides) }
+  end
+
+  describe "instance methods" do
+    it '#unique_mechanics' do
+      @six_flags = AmusementPark.create!(name: 'Six Flags', admission_cost: 75)
+      @hurler = @six_flags.rides.create!(name: 'The Hurler', thrill_rating: 7, open: true)
+      @scrambler = @six_flags.rides.create!(name: 'The Scrambler', thrill_rating: 4, open: true)
+      @ferris = @six_flags.rides.create!(name: 'Ferris Wheel', thrill_rating: 7, open: false)
+      @dave = Mechanic.create!(name: "Dave Williams", years_experience: 30)
+      @barbara = Mechanic.create!(name: "Barbara Jones", years_experience: 15)
+      @kyle = Mechanic.create!(name: "Kyle Smith", years_experience: 22)
+      @bob = Mechanic.create!(name: "Bob Randall", years_experience: 65)
+      MechanicRide.create!(ride_id: @hurler.id, mechanic_id: @dave.id)
+      MechanicRide.create!(ride_id: @scrambler.id, mechanic_id: @dave.id)
+      MechanicRide.create!(ride_id: @ferris.id, mechanic_id: @dave.id)
+      MechanicRide.create!(ride_id: @hurler.id, mechanic_id: @barbara.id)
+      MechanicRide.create!(ride_id: @scrambler.id, mechanic_id: @barbara.id)
+      MechanicRide.create!(ride_id: @scrambler.id, mechanic_id: @kyle.id)
+      MechanicRide.create!(ride_id: @ferris.id, mechanic_id: @kyle.id)
+
+      # This method returns a uniuqe list of mechanics working on rides within the designated park
+      expect(@six_flags.unique_mechanics).to eq([@dave, @barbara, @kyle])
+      expect(@six_flags.unique_mechanics).not_to include(@bob)
+    end
   end
```

Comment:
Great job. I always like to see to different object calling on the instance method but I think this is ok also.

```ruby
expect(@six_flags.unique_mechanics).to eq([@dave, @barbara, @kyle])
expect(@another_park.unique_mechanics).to eq([@dave, @bob])
```

This helps verify that your logic is dynamic and not just:

```ruby
def unique_mechanics
  [@dave, @barbara, @kyle]
end
```